### ISSUE TITLE
[1.12] Crash using crafting table

### ISSUE THREAD

**Issue by Utildayael** [blay09/CookingForBlockheads#458]
So this was working wonderfully for a very long time and somehow I upset the gods. >.<

**Minecraft:** 1.12
**Pack:** MC Eternal
**CookingForBlockheads_1.12.2-6.5.0.jar**

Reproduces same error every time now. Try to open crafting station [counter?] and game crashes back to start menu. It does not crash to desktop thankfully!

**Error:**
```
Description: Ticking player

java.lang.ArrayIndexOutOfBoundsException: 27
    at net.blay09.mods.cookingforblockheads.api.capability.KitchenItemProvider.getSimulatedUseCount(KitchenItemProvider.java:37)
    at net.blay09.mods.cookingforblockheads.api.capability.DefaultKitchenItemProvider.findSource(DefaultKitchenItemProvider.java:45)
    at net.blay09.mods.cookingforblockheads.registry.CookingRegistry.findAnyItemStack(CookingRegistry.java:271)
    at net.blay09.mods.cookingforblockheads.registry.CookingRegistry.findSourceCandidates(CookingRegistry.java:285)
    at net.blay09.mods.cookingforblockheads.registry.CookingRegistry.getRecipeStatus(CookingRegistry.java:331)
    at net.blay09.mods.cookingforblockheads.container.ContainerRecipeBook.findAndSendItemList(ContainerRecipeBook.java:214)
    at net.blay09.mods.cookingforblockheads.container.ContainerRecipeBook.detectAndSendChanges(ContainerRecipeBook.java:146)
    at net.minecraft.entity.player.EntityPlayerMP.onUpdate(EntityPlayerMP.java:324)
    at net.minecraft.world.World.updateEntityWithOptionalForce(World.java:1996)
    at net.minecraft.world.WorldServer.updateEntityWithOptionalForce(WorldServer.java:832)
    at net.minecraft.world.World.updateEntity(World.java:1958)
    at net.minecraft.world.WorldServer.tickPlayers(WorldServer.java:642)
    at net.minecraft.world.World.updateEntities(World.java:1737)
    at net.minecraft.world.WorldServer.updateEntities(WorldServer.java:613)
    at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:767)
    at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:668)
    at net.minecraft.server.integrated.IntegratedServer.tick(IntegratedServer.java:185)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
    at java.lang.Thread.run(Unknown Source)
```

**Full Error report:**
https://pastebin.com/4CGht1TU

Appreciate it! Your mod makes cooking in Minecraft tolerable. :)

**Comment by BlayTheNinth** [comment]
Is `largeCounters` enabled in the config and if so, is the the same on both server and client? You may also have to re-place the kitchen counters if you've changed the value.

**Comment by Utildayael** [comment]
Actually ya I turned on large counters a week or so ago since the storage is so small. I was hoping it would make fridge bigger too. With Pam’s there are like 60 crops + another 20 or so trees and then all the Misc stuff I need a billion slots!

This is SP. I’ll disable the setting and replace all the CFB pieces to make sure it’s all reset. 

Will report back later today. Thanks!

**Comment by Utildayael** [comment]
That seems to have been the problem. Oddly when I looked at the config it was switch back to NOT have largeCounters active. Not sure if this was due to recent pack update and maybe he overwrote the config. Grr.

I did notice my cabinets had 6 rows [I think] when I logged in and down to just 3 rows after re-placing them so it must have saved its state from when config WAS on and maybe I tossed some stuff in to fill it past the first 3 rows so it tried to reference a row that was not supposed to exist? Dunno. I was doing a lot of cooking yesterday and the one cabinet I have with ingredients in it I know I tossed a bunch of stuff in it. [rest is in fridges]

Hope that helps but seems core problem is modpack changed config while cabinets existed in the world with prior large setting.

Thanks for the quick reply!!
